jazz_assets/test/test.js

```diff
@@ -0,0 +1,768 @@
+      const validateCreatePayload = sinon.stub(validateutils, "validateCreatePayload").resolves(event.body);
+      const createNewAsset = sinon.stub(index, "createNewAsset").rejects(err);
+      index.processAssetCreation(event.body, assetTable)
+        .catch(error => {
+          expect(error).to.include(err);
+          sinon.assert.calledOnce(validateCreatePayload);
+          sinon.assert.calledOnce(createNewAsset);
+          validateCreatePayload.restore();
+          createNewAsset.restore();
+        });
+    });
+  });
+
+  describe('processAssetSearch', () => {
+    it("should successfuly process asset serach", () => {
+      var input_data = {
+        service: event.body.service,
+        domain: event.body.domain,
+        environment: event.body.environment
+      }
+      var resObj = {
+        data: event.body,
+        input: input_data
+      };
+      const validateSearchPayload = sinon.stub(validateutils, "validateSearchPayload").resolves(event.body);
+      const postSearch = sinon.stub(index, "postSearch").resolves(resObj);
+      index.processAssetSearch(event.query, assetTable)
+        .then(res => {
+          expect(res).to.include(resObj);
+          sinon.assert.calledOnce(validateSearchPayload);
+          sinon.assert.calledOnce(postSearch);
+          validateSearchPayload.restore();
+          postSearch.restore();
+        });
+    });
+
+    it("should indicate error while processing asset serach", () => {
+      const validateSearchPayload = sinon.stub(validateutils, "validateSearchPayload").resolves(event.body);
+      const postSearch = sinon.stub(index, "postSearch").rejects(err);
+      index.processAssetSearch(event.query, assetTable)
+        .catch(error => {
+          expect(error).to.include(err);
+          sinon.assert.calledOnce(validateSearchPayload);
+          sinon.assert.calledOnce(postSearch);
+          validateSearchPayload.restore();
+          postSearch.restore();
+        });
+    });
+
+  });
+
+  describe('handler', () => {
+    describe('handler with rejected genericInputValidation', () => {
+      it("should indicate BadRequest error in genericInputValidation if method is undefined", () => {
+        const genericInputValidation = sinon.stub(index, "genericInputValidation").rejects({
+          result: "inputError",
+          message: "method cannot be empty"
+        });
+        index.handler(event, context, (error, data) => {
+          expect(error).to.include('{"errorType":"BadRequest","message":"method cannot be empty"}');
+          sinon.assert.calledOnce(genericInputValidation);
+        });
+        genericInputValidation.restore();
+      });
+
+      it("should indicate internal server error", () => {
+        var resObj = '{"errorType":"InternalServerError","message":"Unexpected Server Error"}';
+        const genericInputValidation = sinon.stub(index, "genericInputValidation");
+        index.handler(event, context,(error, data) => {
+          console.log(error)
```

Comment:
you probably want to remove this. Also in other places if any

Reply:
removed